### PR TITLE
variables: change tidb_rc_read_check_ts to instance level

### LINKED ISSUE #37802: change tidb_rc_read_check_ts to instance level
## Enhancement
System variable `tidb_rc_read_check_ts ` makes the read statements don't get timestamp from `PD` at RC isolation, which improves `TiDB` performance. #32922  
But we meet a new scenario in core bank service. Some service needs to run when `tidb_rc_read_check_ts` is on, some  service needs to run when `tidb_rc_read_check_ts` is off. So, we should change system variable from `GLOBAL/SESSION` to `INSTANCE`. 
Some TiDB instances should run when when `tidb_rc_read_check_ts` is on. You should add it to the configure file of of this tidb, as below.
[instance] 
tidb_rc_read_check_ts = true
Other TiDB instances should do nothing about `tidb_rc_read_check_ts` in the configure file.

### PR DESCRIPTION
**Pull request by TonsnakeLin**
Signed-off-by: TonsnakeLin <lpbgytong@163.com>

### What problem does this PR solve?

Issue Number: close #37802 

Problem Summary:

### What is changed and how it works?
This PR changed system variable `tidb_rc_read_check_ts` from GLOBAL/SESSION to INSTANCE

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by TonsnakeLin** [comment]
@cfzjywxk @ekexium @sticnarf @MyonKeminta  PTAL, thanks!

**Comment by sticnarf** [comment]
Is it possible to still allow it to be set in session and global level? `tidb_rc_read_check_ts` is a published variable, we shouldn't break backward compatibility.

**Comment by ekexium** [comment]
According to the design doc of instance scoped variable, I suppose GLOBAL and INSTANCE scope cannot coexist for a variable. If we want backward compatibility we can't give it INSTANCE scope.

**Comment by TonsnakeLin** [comment]
> Is it possible to still allow it to be set in session and global level? `tidb_rc_read_check_ts` is a published variable, we shouldn't break backward compatibility.

It can be set using `set global tidb_rc_read_check_ts = ON|OFF`.
If you set it using `set session tidb_rc_read_check_ts = ON|OFF`, a warning generated, but the value will be changed

**Comment by cfzjywxk** [comment]
> > Is it possible to still allow it to be set in session and global level? `tidb_rc_read_check_ts` is a published variable, we shouldn't break backward compatibility.
> 
> It can be set using `set global tidb_rc_read_check_ts = ON|OFF`. If you set it using `set session tidb_rc_read_check_ts = ON|OFF`, a warning generated, but the value will be changed

So it means it could still take effect as a session or global variable?

**Comment by TonsnakeLin** [comment]
> > It can be set using `set global tidb_rc_read_check_ts = ON|OFF`. If you set it using `set session tidb_rc_read_check_ts = ON|OFF`, a warning generated, but the value will be changed
> 
> So it means it could still take effect as a session or global variable?
No, It's instance level, but we can still use "set global xxxx", "set session xxxx" , every instance variables is like so.

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -456,6 +456,12 @@ var defaultSysVars = []*SysVar{
 			return BoolToOnOff(config.GetGlobalConfig().Instance.TiDBEnableDDL.Load()), nil
 		},
 	},
+	{Scope: ScopeInstance, Name: TiDBRCReadCheckTS, Value: BoolToOnOff(DefRCReadCheckTS), Type: TypeBool, SetGlobal: func(s *SessionVars, val string) error {
```
Comment (sticnarf) [inline sessionctx/variable/sysvar.go:459]: Should the `Value` be set to the instance config in global config?

Comment (TonsnakeLin) [reply sessionctx/variable/sysvar.go:459]: In fact, I added this instance variable referring to tidb_general_log. It looks simple.
um, what's the difference of this way and global config?

Comment (sticnarf) [reply sessionctx/variable/sysvar.go:459]: I see, it's actually set in `setGlobalVars`. The whole mechanism looks not so consistent, though...

**Review by sticnarf — APPROVED** [review]

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**